openjpa-jdbc/src/main/java/org/apache/openjpa/jdbc/schema/Table.java

```diff
@@ -329,10 +341,19 @@ public Column getColumn(String name) {
         return getColumn(DBIdentifier.newIdentifier(name, DBIdentifierType.COLUMN, true));
     }
 
-    public Column getColumn(DBIdentifier name) {
+    private Column internalGetColumn(DBIdentifier name) {
         if (DBIdentifier.isNull(name) || _colMap == null)
             return null;
-        return _colMap.get(DBIdentifier.toUpper(name));
+        DBIdentifier key = normalizeColumnKey(name);
+        return _colMap.get(key);
+    }
+
+    public Column getColumn(DBIdentifier name) {
+        return internalGetColumn(name);
+    }
+
+    private DBIdentifier normalizeColumnKey(DBIdentifier name) {
+        return DBIdentifier.removeDelimiters(DBIdentifier.toUpper(name, true));
```

Comment:
@rmannibucau about applying removeDelimiters only when needed:
my original fix was to handle this only in SchemaTool, but just adding the new test case showed that  the situation is more complex.
Table class does not have a reference to the DBDictionary and we can't change the signature of all methods, it is kind of Public API that can be overridden by custom DBDictionary implementations

Tests are also failing if you switch to `DBIdentifier.toUpper(DBIdentifier.removeDelimiters(name), true);`

IMHO we could also probably clean up some of those methods in order to reduce useless allocation, but it can be a separate follow up patch.

Reply:
the identifier should always (i know a few cases where it is not the case but shouldnt impact you or means it must be fixed) be created from the dict so just inject that state in the identifier? Makes sense to me to know if the identifier had been quoted or not and does not require a single signature change in methods.

Reply:
> Makes sense to me to know if the identifier had been quoted or not and does not require a single signature change in methods
I would like this change but it is not trivial, as I see that sometimes we are manipulating the internal `_name` of DBIdentifier, and tracking when it is already delimited or not seems to me an hard task.
It can be done in a follow up patch. I will be happy to follow up in this direction

Reply:
It is not really about the overhead (should be a "contains" in terms of complexity for most cases) but more about being fully deterministic and not "random".

Why is it hard?
Tables can be created in:
- openjpa-jdbc/src/main/java/org/apache/openjpa/jdbc/sql/DBDictionary.java#newTable
- openjpa-jdbc/src/main/java/org/apache/openjpa/jdbc/sql/DBDictionary.java#getValidColumnName
- org.apache.openjpa.jdbc.schema.Schema#addTable
- org.apache.openjpa.jdbc.schema.SchemaGroup#newTable

Now if you take 1 more level you always have the dictionnary handy so if simpler we can just add the dict in the table.

Alternative I had in mind was just to handle it in org.apache.openjpa.jdbc.sql.DBDictionary#getValidColumnName(org.apache.openjpa.jdbc.identifier.DBIdentifier, org.apache.openjpa.jdbc.schema.Table, boolean) in herddb dict and set the flag there.
It is not about tracking it is delimited or not but tracking that the dict is in delimited mode or not, then you would do 
 if identifier.delimited then removeDelimited() else noop.